### DISCUSSION TITLE
is it possible and does it make sense to have one autopilot repo manage multiple argo installations (in different clusters)?

### DISCUSSION THREAD

**Discussion by sebastian-popa** [argoproj-labs/argocd-autopilot#536] [Q&A] [answered]
Quick context: 
- different kube clusters (one per environment - dev, staging, prod)
- different argo installs (autopilot or not) - these so one can upgrade dev argo without interfering with prod argo, for example

Question is if such an arrangement makes sense, and if it's possible while using the bootstrapping from autopilot. Thinking basically if one can have bootstrap/dev, bootstrap/staging and so subfolders, and similar for the apps folder.

Thanks!

**Comment by ghost** [comment, marked as answer]
after installing an autopilot repo on two different clusters (`cluster-A` and `cluster-B`), using two different repositories (`repo-A` and `repo-B`), it is possible to then go to the `autopilot-bootstrap` application in one of those clusters (`cluster-B`), and manually edit its `spec.source` fields to actually reference the other repository (`repo-A`). after doing that - you will have two different clusters with identical resources (argo-cd + whatever other applications you have set up inside `repo-A`). `repo-B` will not be used for anything at this point.
i am not entirely sure what would be the use-case. but if you want two identical clusters set up that way - it should work.

another option, which i think is *not* what you had in mind, is to set up the two autopilot repositories inside a single git repository. just under different paths. so run something like
```shell
argocd-autopilot repo bootstrap --repo https://github.com/some-org/single-repo/path-a --context cluster-a
```
to install it it in cluster-A, and then
```shell
argocd-autopilot repo bootstrap --repo https://github.com/some-org/single-repo/path-b --context cluster-b
```
to install in cluster-B.

now you will have two different autopilot-managed clusters, using different paths inside the same repo. so technically this is what you asked for, but for almost any other purpose they are just as well in separate repositories.

**Reply by sebastian-popa** [reply]
thanks @noam-codefresh - I think the second option is what I was looking for. I'm trying to keep the number of github repo down, and since "everything else" uses one repo for all environments, I'd like autopilot to do the same.
